File: src/camera_subscriber.cpp
#include <cv_bridge/cv_bridge.h>
#include <image_transport/image_transport.h>
#include <ros/ros.h>
#include <sensor_msgs/Image.h>

#include <opencv2/highgui/highgui.hpp>

void imgCallback(const sensor_msgs::Image::ConstPtr &msg) {
  try {
    cv::imshow("Image", cv_bridge::toCvShare(msg, "bgr8")->image);
    cv::waitKey(1);
  } catch (cv_bridge::Exception &e) {
    ROS_ERROR("Could not convert from '%s' to 'bgr8'.", msg->encoding.c_str());
  }
}

int main(int argc, char **argv) {
  ros::init(argc, argv, "camera_subscriber");  // Node initialization
  ros::NodeHandle n;                           // Nodehandler
  ros::Subscriber subLeftImg =
      n.subscribe("camera/image", 10, imgCallback);  // Subscriber
  ros::spin();                                       // Node execution
  return 0;
}

File: scripts/camera_subscriber.py
#!/usr/bin/env python
"""
Simple Camera subscriber
"""
import cv2
import rospy
from cv_bridge import CvBridge, CvBridgeError
from sensor_msgs.msg import Image
bridge = CvBridge()


def imgCallback(img):
    try:
        img = bridge.imgmsg_to_cv2(img, "bgr8")
    except CvBridgeError as e:
        print(e)
    else:
        cv2.imshow('Image', img)
        cv2.waitKey(1)


def main():
    rospy.init_node('image_listener')
    rospy.Subscriber("camera/image", Image, imgCallback)
    rospy.spin()


if __name__ == "__main__":
    main()
